docs/docs/build-caching.md

```diff
@@ -3,7 +3,71 @@ title: Build Caching
 issue: https://github.com/gatsbyjs/gatsby/issues/8103
 ---
 
-This is a stub. Help our community expand it.
+Plugins can cache data as json objects and retrieve them on consecutive builds.
```

Comment:
We should also mention a couple other things 

- any nodes created by source/transformer plugins are cached.
- gatsby-plugin-sharp caches built thumbnails